Add cooperative groups (#3478)
ROCm-Developer-Tools/HIP: docs/reference/cooperative_groups_reference.rst

> .. meta::

Reviewer: See the ticket. 
Included a document with changes to apply.

Author: Fixed.

.. meta::

>   :description: This chapter lists new types and device API wrappers related to Cooperative Group

Reviewer: ```suggestion
  :description: This topic lists types and device API wrappers related to Cooperative Group
```

>   :description: This chapter lists new types and device API wrappers related to Cooperative Group
>                 feature, which the programmer can directly use in his kernel(s) in order to
>                 make use of this feature.

Reviewer: ```suggestion
  :description: This chapter lists  types and device API wrappers related to the Cooperative Group
                feature. Programmers can directly use them in their kernels to make use of this feature.
```

  :keywords: AMD, ROCm, HIP, cooperative groups

*******************************************************************************

> Cooperative Groups API

Reviewer: Looks like this content is coming from elsewhere. Some comments.
Change this line everywhere: This function is implemented on Linux, under developement on Windows.
To: This function is implemented on Linux and is under development on Microsoft Windows.

Be gender neutral - see this example:
Instead, he should construct it through this API function.
It should be:
Instead, they should construct it through this API function.

Author: This is in clr. Updating the section too:

https://github.com/ROCm/clr/pull/81/commits

Author: Fixed.

.. meta::
  :description: This chapter lists  types and device API wrappers related to the Cooperative Group
                feature. Programmers can directly use them in their kernels to make use of this feature.
  :keywords: AMD, ROCm, HIP, cooperative groups

*******************************************************************************
Cooperative Groups API
*******************************************************************************

> 

Reviewer: Perhaps we should include something like this so we have a cross reference between reference and How To.
this should be a standard. 

This topic lists and describes the cooperative groups functions. For more information about how to use cooperative groups, see https://advanced-micro-devices-demo--3478.com.readthedocs.build/projects/HIP/en/3478/how-to/cooperative_groups.html.

Perhaps a link in the How to to the Reference, too.

Author: Added references.
